nixos/modules/services/monitoring/telegraf.nix

```diff
@@ -29,7 +29,9 @@ in {
       extraConfig = mkOption {
         default = {};
         description = "Extra configuration options for telegraf";
-        type = types.attrs;
+        type = types.attrsOf (types.either
+          types.attrs
+          (types.attrsOf (types.listOf types.attrs)));
```

Comment:
A config I'm deploying looks like
```
[agent]
metric_batch_size = 500

[global_tags]
customer = "awake"
serial = "S268672X0C19144"

[inputs]

[[inputs.cpu]]

[[inputs.disk]]

[[inputs.diskio]]

[[inputs.jolokia2_agent]]
fielddrop = ["*.EventType", "*.FifteenMinuteRate", "*.FiveMinuteRate", "*.MeanRate", "*.OneMinuteRate", "*.RateUnit", "*.LatencyUnit", "*.50thPercentile", "*.75thPercentile", "*.95thPercentile", "*.98thPercentile", "*.99thPercentile", "*.999thPercentile", "*.Min", "*.Mean", "*.Max", "*.StdDev"]
name_prefix = "kafka_"
tagexclude = ["jolokia_agent_url"]
urls = ["http://172.18.0.10:9998/jolokia"]
```

I would read that as:
```
types.attrsOf (types.either
          types.attrs
          (types.listOf types.attrs));
```

Am I reading that wrong?

Reply:
I think in your example, inputs, for example, corresponds to
```nix
extraConfig = {
  inputs = {
    cpu = [{ }];
    disk = [{ }];
    diskio = [{ }];
    jolokia2_agent = [{
      fielddrop = [ "*.EventType" "*.FifteenMinuteRate" "*.FiveMinuteRate" "*.MeanRate" "*.OneMinuteRate" "*.RateUnit" "*.LatencyUnit" "*.50thPercentile" "*.75thPercentile" "*.95thPercentile" "*.98thPercentile" "*.99thPercentile" "*.999thPercentile" "*.Min" "*.Mean" "*.Max" "*.StdDev" ];
      name_prefix = "kafka_";
      tagexclude = [ "jolokia_agent_url" ];
      urls = [ "http://...jolokia" ];
    }];
  };
};
```
Note that `inputs` (or `outputs`, or `aggregators`, etc) are the inner `attrsOf` (I think). I am confirming that this outputs valid config right now.

Reply:
I yeah, you're correct, sorry this is confusing :D